lib/SPIRVProducerPass.cpp

```diff
@@ -5292,6 +5321,9 @@ void SPIRVProducerPassImpl::GenerateInstruction(Instruction &I) {
       Ops << ST->getValueOperand();
     }
     RID = addSPIRVInst(spv::OpStore, Ops);
+
+    DecoratePtrIfNonUniform(ptr);
```

Comment:
I think this needs to be added for any memory operation (e.g. atomics, OpCopyMemory, etc).

Reply:
Maybe it's possible to refactor to the end of GenerateInstruction.

Reply:
I have refactored it to work on the operands of each instruction at the beginning of GenerateInstruction.